lib/toast/toast.test.tsx

```diff
@@ -0,0 +1,166 @@
+import React from 'react';
+
+import Toast from './toast';
+
+import { NAMESPACE } from '../utilities/ts/constants';
+
+import { mount } from 'enzyme';
+
+let wrapper;
+
+describe(`<Toast />`, () => {
+  beforeEach(() => {
+    wrapper = mount(
+      <Toast id="foo" heading="Bar" message="Baz" duration={1000} />
+    );
+  });
+
+  it(`Handles props.className`, () => {
+    expect(wrapper.getDOMNode().getAttribute(`class`)).toBe(
+      `${NAMESPACE}-toast mounted`
+    );
+
+    wrapper.setProps({ className: `baz` });
+    expect(wrapper.getDOMNode().getAttribute(`class`)).toBe(
+      `${NAMESPACE}-toast mounted baz`
+    );
+
+    wrapper.setProps({ className: `baz qui` });
+    expect(wrapper.getDOMNode().getAttribute(`class`)).toBe(
+      `${NAMESPACE}-toast mounted baz qui`
+    );
+  });
+
+  it(`Handles props.duration`, done => {
+    setTimeout(() => {
+      expect(wrapper.getDOMNode().getAttribute(`class`)).toBe(
+        `${NAMESPACE}-toast mounted`
+      );
+    }, 500);
+
+    setTimeout(() => {
+      expect(wrapper.getDOMNode().getAttribute(`class`)).toBe(
+        `${NAMESPACE}-toast`
+      );
+    }, 1000);
+
+    setTimeout(() => {
+      wrapper.update();
+
+      expect(wrapper).toMatchObject({});
+      done();
+    }, 2000);
```

Comment:
why 500, 1000 and 2000? Is 500ms the hardcoded value from the toast default behaviour (toast.tsx , l68), and 1000 the specific duration value for this test? what about 2000ms?

What do these values represent?

Also, what happens to the wrapper at, lets say, 1800ms?  would it match {} as well?

Reply:
The `setTimeout()` at these values provides enough time for `<Toast />` transitions to occur, we then ping the `wrapper` to test. Does this make sense, @victorliafook?